src/core/agent_space_interaction.jl

```diff
@@ -40,30 +40,39 @@ kill_agent!(id::Integer, model) = kill_agent!(model[id], model)
     genocide!(model::ABM)
 Kill all the agents of the model.
 """
-genocide!(model::ABM) =
+genocide!(model::ABM{A,<:DiscreteSpace}) where {A} = _genocide!(model)
+genocide!(model::ABM{A,Nothing}) where {A} = _genocide!(model)
```

Comment:
Both methods actually do the same thing, why dispatch on space here?

Reply:
I kept getting ambiguous errors if we just set this as `genocide!(model)`. I fixed a bunch of continuous space issues after this though, so maybe those errors are dropped now. Give me a sec an I'll revert these to see...

Reply:
If it resolves ambiguity its okay.

Reply:
Yep. Here:
```julia
genocide!: Error During Test at /mnt/turtle/repos/Agents.jl/test/api_tests.jl:271
  Got exception outside of a @test
  MethodError: genocide!(::AgentBasedModel{Agent8,ContinuousSpace{typeof(Agents.defvel),Tuple{Int64,Int64}},typeof(fastest),Nothing}) is ambiguous. Candidates:
    genocide!(model::AgentBasedModel{A,#s60,F,P} where P where F where #s60<:ContinuousSpace) where A in Agents at /mnt/turtle/repos/Agents.jl/src/core/continuous_space.jl:214
    genocide!(model::AgentBasedModel) in Agents at /mnt/turtle/repos/Agents.jl/src/core/agent_space_interaction.jl:43
  Possible fix, define
    genocide!(::AgentBasedModel{A,S,F,P} where P where F where S<:ContinuousSpace) where A<:AbstractAgent
```

Call is:
```julia
  space2d = ContinuousSpace(2; periodic=true, extend=(1, 1))    
  model = ABM(Agent8, space2d)    
  attributes = (f1=true,f2=1)    
  for _ in 1:10 add_agent!(model, attributes...) end    
  genocide!(model)    
```
Ambiguous methods:
```julia
genocide!(model::ABM) = ... # agent_space_interaction.jl#43

genocide!(model::ABM{A,<:ContinuousSpace}) where {A} = ... # continuous_space.jl#214
```

How best to rectify that?

Reply:
Maybe adding type info for genocide in discrete space?
```julia
# agent_space_interaction.jl#43
genocide!(model::ABM{A,<:DiscreteSpace}) where {A}
    for a in allagents(model)
        kill_agent!(a, model)
    end
end
```

Reply:
But then we don't have genocide for `Nothing`.

Reply:
yeah, right!

On a different note, isn't massacre better than genocide?

Reply:
```julia
genocide!(model)
decimate!(model) #where agent_ids.%10==0
massacre!(model) #where 20 < agent_ids < nagents(model)*0.1
```

Reply:
Let's minimize breaking changes by keeping `genocide!`. Other than that, I don't care as long as the name remains funny :D